client/blocks/site/index.jsx

```diff
@@ -68,6 +73,33 @@ class Site extends React.Component {
 		this.props.onMouseLeave( event, this.props.site.ID );
 	};
 
+	onViewSiteClick = event => {
+		const { isPreviewable, siteSuffix } = this.props;
+
+		if ( ! isPreviewable ) {
+			this.trackMenuItemClick( 'view_site_unpreviewable' );
+			this.props.recordGoogleEvent( 'Sidebar', 'Clicked View Site | Unpreviewable' );
+			return;
+		}
+
+		if ( event.altKey || event.ctrlKey || event.metaKey || event.shiftKey ) {
```

Comment:
The purpose of this code is to implement a neat feature for a `<a href>` link: if the link is clicked, JavaScript will open a preview modal with iframe. The `href` navigation is not performed. However, when the link is clicked with Ctrl-click, the intent is to open the link in new tab. Then the JavaScript doesn't intercept and the default browser behavior for `href` click is performed. If the `href` contains the site's frontend URL, it opens the site in new tab!

For `<button>`, this code is useless. But you can implement the behavior above by switching back to `<a>`. The linter will not yell at you any more, because the `a` element will have a valid `href` attribute.